langchain/src/prompts/chat.ts

```diff
@@ -255,17 +255,27 @@ export class ChatPromptTemplate
   }
 
   static fromPromptMessages(
-    promptMessages: BaseMessagePromptTemplate[]
+    promptMessages: (BaseMessagePromptTemplate | ChatPromptTemplate)[]
   ): ChatPromptTemplate {
+    const flattenedMessages = promptMessages.reduce(
```

Comment:
Whenever partial is implemented for this class this needs to be adjusted accordingly